src/gluonts/maybe.py

```diff
@@ -76,25 +126,31 @@ def do(val: Optional[T], fn: Callable[[T], U]) -> Optional[T]:
     >>> do(None, print)
 
     """
-    if val is not None:
-        fn(val)
-
-    return val
+    return box(val).do(fn).unbox()
 
 
-def map(val: Optional[T], fn: Callable[[T], U]) -> Optional[U]:
+def map(
```

Comment:
I’m wondering: why is this called `map` and not for example `apply`? `map` sounds more like the built-in `map` that deals with iterables. (Same for `map_or` and `map_or_else`.)

Reply:
I guess I need to refresh my category theory!